app/models/groups_importer.rb

```diff
@@ -17,20 +17,30 @@ def import_from_csv(file_path)
     # Widgets:
     #
     widgets = {
-      'pie' => API::V1::FindOrCreateWidget.call('name' => 'pie'),
-      'line' => API::V1::FindOrCreateWidget.call('name' => 'line'),
-      'bar' => API::V1::FindOrCreateWidget.call('name' => 'bar'),
-      'choropleth' => API::V1::FindOrCreateWidget.call('name' => 'choropleth')
+      'pie' => API::V1::FindOrUpsertWidget.call(name: 'pie'),
+      'line' => API::V1::FindOrUpsertWidget.call(name: 'line'),
+      'bar' => API::V1::FindOrUpsertWidget.call(name: 'bar'),
+      'choropleth' => API::V1::FindOrUpsertWidget.call(name: 'choropleth')
     }
 
-    CSV.foreach(file_path, headers: true, converters: :numeric) do |row|
+    CSV.foreach(file_path, headers: true, converters: :numeric).with_index do |row, index|
+
       row_data = row.to_hash.transform_keys! { |key| key.to_s.downcase }
+      next unless valid_row?(row_data, index)
```

Comment:
I've found that some rows with bad data( I presume that some values for some columns were missed ) ending up with Chinese strings into cells that are supposed to be English.